arithmetization/src/test/java/net/consensys/linea/zktracer/module/blsdata/BlsPairingCheckTest.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package net.consensys.linea.zktracer.module.blsdata;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static net.consensys.linea.zktracer.Fork.isPostPrague;
+import static net.consensys.linea.zktracer.module.blsdata.BlsTestUtils.LARGE_POINTS;
+import static net.consensys.linea.zktracer.module.blsdata.BlsTestUtils.SMALL_POINTS;
+import static net.consensys.linea.zktracer.module.blsdata.BlsTestUtils.VALID_G1_POINT;
+import static net.consensys.linea.zktracer.module.blsdata.BlsTestUtils.VALID_G2_POINT;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import net.consensys.linea.UnitTestWatcher;
+import net.consensys.linea.reporting.TracerTestBase;
+import net.consensys.linea.testing.BytecodeCompiler;
+import net.consensys.linea.testing.BytecodeRunner;
+import net.consensys.linea.testing.ToyAccount;
+import net.consensys.linea.zktracer.opcode.OpCode;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.Wei;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+@ExtendWith(UnitTestWatcher.class)
+public class BlsPairingCheckTest extends TracerTestBase {
+
+  @ParameterizedTest
+  @MethodSource("blsPairingCheckSource")
+  void testBlsPairingCheckTest(
```

Comment:
It would be good to add documentation. What is it we are checking here ? It doesn't have to be long but some description would help in the future e.g.
```java
/**
* The following test tests the BLS_DATA module's ability to recognize malformed points
* and, for well formed points, to offload curve membership and subgroup membership tests
* to gnark, in a way which is in accordance with the EVM and the return bit of the underlying
* <b>CALL</b>.
*/

Reply:
I would suggest we explicitly document all tests.